test/volumes-list-params-validation.test.js

```diff
@@ -0,0 +1,524 @@
+                });
+
+                CLIENT.get(query, function onListVolumes(err, req, res, obj) {
+                    t.ok(err,
+                        'listing volumes with invalid predicate should error');
+                    t.equal(err.restCode, 'InvalidArgument',
+                        'error should InvalidArgument');
+                    t.end();
+                });
+            });
+    });
+
+    // copied from sdc-volapi test/integration/list-with-params.test.js
+    function snowflakeName(strName) {
+        switch (strName) {
+            case 'snowflakeName0':
+                return snowflakeName0;
+            case 'snowflakeName1':
+                return snowflakeName1;
+            case 'snowflakeName2':
+                return snowflakeName2;
+            default:
+                return 'unknown volume';
+        }
+    }
+
+    // copied from sdc-volapi test/integration/list-with-params.test.js
+    function shouldFind(t, volumes, expected, notExpected, expectedNumber) {
+        var foundVolumes = [];
+        var idx;
+
+        t.ok(Array.isArray(volumes), 'response body should be an array');
+        if (expectedNumber !== undefined) {
+            t.equal(volumes.length, expectedNumber, expectedNumber +
+                ' volume(s) should be included in the response body');
+        }
+
+        volumes.forEach(function checkVolume(vol) {
+            switch (vol.name) {
+                case snowflakeName0:
+                    foundVolumes.push('snowflakeName0');
+                    break;
+                case snowflakeName1:
+                    foundVolumes.push('snowflakeName1');
+                    break;
+                case snowflakeName2:
+                    foundVolumes.push('snowflakeName2');
+                    break;
+                default:
+                    foundVolumes.push('unknownName');
+                    break;
+            }
+        });
+
+        for (idx = 0; idx < expected.length; idx++) {
+            t.ok(foundVolumes.indexOf(expected[idx]) !== -1,
+                'should have found ' + snowflakeName(expected[idx]));
+        }
+
+        for (idx = 0; idx < notExpected.length; idx++) {
+            t.ok(foundVolumes.indexOf(notExpected[idx]) === -1,
+                'should not have found ' + snowflakeName(notExpected[idx]));
+        }
+
+        return (foundVolumes);
+    }
+
+    test('create some test volumes', function (tt) {
+        var testVolumeObjects = [
+            {
+                name: snowflakeName0,
+                size: 10240,
+                type: 'tritonnfs'
+            },
+            {
+                name: snowflakeName1,
+                size: 102400,
+                type: 'tritonnfs'
+            },
+            {
+                name: snowflakeName2,
+                size: 1024000,
+                type: 'tritonnfs'
+            }
+        ];
+
+        testVolumeObjects.forEach(function createVolume(newVol) {
```

Comment:
nit: does the output of each test created by the `createVolume` callback ends up being interleaved? If so, would it help to use a `vasync.forEachPipeline`?